### PR TITLE
Update-838: rack-to-rack docs

### PR DESCRIPTION
**Pull request by Twsouza**
It will include:

Update-838: rack-to-rack docs (#583)

### CODE REVIEW

**Comment by Twsouza** [comment]
@lucasmacedot some of the changes you made is failing in the E2E tests, can you take a look at it?

```
Error: one of route_table_id, vpc_id, subnet_id, gateway_id, filters, or tags must be assigned

  with module.system.module.cluster.data.aws_route_table.public,
  on .terraform/modules/system/terraform/cluster/aws/network.tf line 79, in data "aws_route_table" "public":
  79: data "aws_route_table" "public" {


Error: one of route_table_id, vpc_id, subnet_id, gateway_id, filters, or tags must be assigned

  with module.system.module.cluster.data.aws_route_table.private[0],
  on .terraform/modules/system/terraform/cluster/aws/network.tf line 173, in data "aws_route_table" "private":
 173: data "aws_route_table" "private" {
```

**Comment by lucasmacedot** [comment]
Yeah, will fix it

**Comment by lucasmacedot** [comment]
Hey @Twsouza , I just commented a fix as a suggestion.
You prefer it this way? Or you want me to open a PR?

File: terraform/system/aws/main.tf [hunk 1]
```diff
@@ -65,6 +65,8 @@ module "cluster" {
   tags                     = local.tag_map
   vpc_cni_version          = var.vpc_cni_version
   vpc_id                   = var.vpc_id
+  private_subnets_ids      = split(",", var.private_subnets_ids)
+  public_subnets_ids       = split(",", var.public_subnets_ids)
```
[the remark this answers is not in the document]
Comment (lucasmacedot) [reply terraform/system/aws/main.tf:69]: The compact function removes any emtpy string of a array: https://developer.hashicorp.com/terraform/language/functions/compact

**Comment by Twsouza** [comment]
@lucasmacedot it's okay, I'll rerun the tests

File: terraform/cluster/aws/network.tf [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
     Name = var.name
     Rack = var.name
   })
   vpc_id = var.vpc_id == "" ? aws_vpc.nodes[0].id : var.vpc_id
+
+  private_subnets_ids  = length(var.private_subnets_ids) == 0 ? aws_subnet.private[*].id : var.private_subnets_ids
+  private_route_tables = length(var.private_subnets_ids) == 0 ? aws_route_table.private[*].id : data.aws_route_table.private[*].id
+
+  public_subnets_ids  = length(var.public_subnets_ids) == 0 ? aws_subnet.public[*].id : var.public_subnets_ids
+  public_route_table  = length(var.public_subnets_ids) == 0 ? aws_route_table.public[0].id : data.aws_route_table.public.id
```
Comment (lucasmacedot) [inline terraform/cluster/aws/network.tf:12]:
```suggestion
  public_route_table  = length(var.public_subnets_ids) == 0 ? aws_route_table.public[0].id : data.aws_route_table.public[0].id
```

File: terraform/cluster/aws/network.tf [hunk 3]
```diff
@@ -70,7 +76,12 @@ resource "aws_subnet" "public" {
   }
 }
 
+data "aws_route_table" "public" {
+  subnet_id = var.public_subnets_ids[0]
+}
```
Comment (lucasmacedot) [inline terraform/cluster/aws/network.tf:81]:
```suggestion
data "aws_route_table" "public" {
  count     = length(var.public_subnets_ids)
  subnet_id = var.public_subnets_ids[count.index]
}
```

**Comment by lucasmacedot** [comment]
Sorry, these tests are not on the make test :<

From what I have gathered, there are 3 alternatives:
- No VPC
- Specifying VPC and IGW
- Specifying VPC, IGW and subnets

I ran the plan on all 3, and they worked, don't know if there are any others

**Comment by lucasmacedot** [comment]
I don't think I can currently add the test for the existing subnets on the github workflow, because there's the data for the route_tables, so the route_tables would have to exist on the AWS account that the github workflows run

**Comment by Twsouza** [comment]
@lucasmacedot can you open a PR for this?

**Comment by lucasmacedot** [comment]
@Twsouza https://github.com/convox/convox/pull/587

**Comment by Twsouza** [comment]
@lucasmacedot I need to release the docs update, I reverted your changes but you are free to open a PR with the changes once they are working.

**Comment by lucasmacedot** [comment]
That's alright!

**Pull request merged**